controllers/devworkspace/solver/che_routing.go

```diff
@@ -342,14 +361,33 @@ func (c *CheRoutingSolver) getGatewayConfigsAndFillRoutingObjects(cheCluster *ch
 	return configs, nil
 }
 
-func (c *CheRoutingSolver) getInfraSpecificExposer(cheCluster *chev2.CheCluster, routing *dwo.DevWorkspaceRouting, objs *solvers.RoutingObjects) (func(info *EndpointInfo), error) {
+func getUsername(c client.Client, namespace string) (string, error) {
+	secret := &corev1.Secret{}
+	err := c.Get(context.TODO(), client.ObjectKey{Name: "user-profile", Namespace: namespace}, secret)
+	if err != nil {
+		return "", err
+	}
+	username := string(secret.Data["name"])
+	return username, nil
```

Comment:
The username here is being collected from the `user-profile` secret. There is a separate che-server PR to allow this: https://github.com/eclipse-che/che-server/pull/508

Another way the username can be collected is by reading the `che.eclipse.org/username` annotation of the user namespace.